osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -340,6 +345,25 @@ private void newItemsPresented()
 
         #region Beatmap management
 
+        /// <summary>
+        /// Whether entering editor mode should be allowed.
+        /// </summary>
+        public virtual bool AllowEditing => true;
+
+        /// <summary>
+        /// Opens beatmap editor with the given beatmap.
+        /// </summary>
+        public void OpenEditor(BeatmapInfo beatmap)
+        {
+            if (!AllowEditing)
+                throw new InvalidOperationException($"Attempted to edit when {nameof(AllowEditing)} is disabled");
+
+            // Forced refetch is important here to guarantee correct invalidation across all difficulties.
+            Beatmap.Value = beatmaps.GetWorkingBeatmap(beatmap, true);
```

Comment:
I've copied this from old song select as is, I'm not 100% sure if this "refetch" has anything to do with how old song select was written.

Reply:
A 30 second blame says "yes it needs to stay".

https://github.com/ppy/osu/pull/27997